### PR TITLE
MM-40880: Sentry crash: MarkChannelAsUnread

### PR DESCRIPTION
**Pull request by agnivade**
Use the correct error variable to prevent crash

The errors.As check has been removed because ThreadStore.Get
returns a nil error if no thread is found and that's how
the application logic is written. So the check was redundant.

```release-note
NONE
```

### CODE REVIEW

File: app/channel_test.go [hunk 1]
```diff
@@ -1992,6 +1993,78 @@ func TestMarkChannelsAsViewedPanic(t *testing.T) {
...
+	require.NoError(t, err)
+
+	mockPreferenceStore := mocks.PreferenceStore{}
+	mockPreferenceStore.On("Get", "userID", model.PreferenceCategoryDisplaySettings, model.PreferenceNameCollapsedThreadsEnabled).Return(&model.Preference{Value: "on"}, nil)
+
+	mockThreadStore := mocks.ThreadStore{}
+	mockThreadStore.On("MarkAllAsReadInChannels", "userID", []string{"channelID"}).Return(nil)
+	mockThreadStore.On("GetMembershipForUser", "userID", "rootID").Return(nil, nil)
+	mockThreadStore.On("MaintainMembership", "userID", "rootID", mock.AnythingOfType("store.ThreadMembershipOpts")).Return(&model.ThreadMembership{}, nil)
+	mockThreadStore.On("Get", "rootID").Return(nil, errors.New("bad error"))
```
Comment (streamer45) [inline app/channel_test.go:2031]: Should we highlight this somehow? Most of the test seems to be boilerplate so it gets hard to follow what is that we are really testing for.

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

File: app/channel_test.go (hunk 1 above)
Comment (agnivade) [reply app/channel_test.go:2031]: Good point :+1:

**Comment by agnivade** [comment]
@amyblais - This is a good candidate for 6.3 ESR.

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

**Comment by amyblais** [comment]
@agnivade Does this need QA review?

**Comment by agnivade** [comment]
@amyblais - No need for that. There's already a unit test.
